test/e2e/branding_test.go

```diff
@@ -0,0 +1,71 @@
+package e2e
+
+import (
+	"k8s.io/apimachinery/pkg/util/wait"
+	"testing"
+	"time"
+
+	operatorsv1 "github.com/openshift/api/operator/v1"
+	consoleapi "github.com/openshift/console-operator/pkg/api"
+	"github.com/openshift/console-operator/pkg/testframework"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Test the customization features of the config map
+
+// the simpler test first would be branding set via the operator config
+func TestOperatorConfigBranding(t *testing.T) {
+	expectedBrand := operatorsv1.BrandDedicated
```

Comment:
Would there be value in:

```golang
// loop and test the valid options?
validBrands := []string{"okd", "ocp", "online", "dedicated", "azure"}
// as well as testing a few invalid?
invalidBrands := []string{"redhat", "ninnyhammers", ""}
```
How much would that increase our run time, and would be worth the assertions?

Reply:
So I keep trying this and I get through 4 of the 5 options but the fifth always fails with:
`Operation cannot be fulfilled on consoles.operator.openshift.io "cluster": the object has been modified; please apply your changes to the latest version and try again`
I change the order and still the first 4 pass then it errors out.
Additionally I added an invalid option and it does fail with a proper pattern statement:
`spec.customization.brand in body should match '^$|^(ocp|origin|okd|dedicated|online|azure)$`
 but I am not sure writing logic to catch it and not cause the test to fail is simple.

Reply:
likely that just means your tests are too fast.  order doesn't really matter, just need a sleep in between them.